pkg/prometheus/statefulset.go

```diff
@@ -315,6 +314,10 @@ func makeStatefulSetSpec(
 	// Prometheus may take quite long to shut down to checkpoint existing data.
 	// Allow up to 10 minutes for clean termination.
 	terminationGracePeriod := int64(600)
+	cg, err := NewConfigGenerator(logger, &p, false)
```

Comment:
I would pass to the function the config generator already instantiated by the controller.

Reply:
This ended up generating a bit more work, due to having to re-write the parameters of the function but I agree that only having one ConfigGenerator is the way to go